tests/main/snapd-reexec-snapd-snap/task.yaml

```diff
@@ -12,9 +12,15 @@ execute: |
         exit 0
     fi
 
-    echo "Enable installing the snapd snap"
+    echo "Enable installing the snapd snap, this happens automatically"
     snap set core experimental.snapd-snap=true
-    snap install --edge snapd
+    for _ in $(seq 20); do
```

Comment:
is this kind of loop something for the new retry-tool? or is that for other uses (I haven't followed the details of that) cc @zyga

Reply:
Indeed, I updated the code to use it and also make it a bit smarter (i.e. it now attaches itself to the change for extra debug juice)